yoti_python_sdk/tests/dynamic_sharing_service/policy/test_dynamic_policy_builder.py

```diff
@@ -0,0 +1,119 @@
+
+
+def test_remember_me():
+    policy = DynamicPolicyBuilder().with_wanted_remember_me().build()
+
+    assert policy["wanted_remember_me"]
+
+
+def test_build_with_simple_attributes():
+    builder = DynamicPolicyBuilder()
+    builder.with_family_name()
+    builder.with_given_names()
+    builder.with_full_name()
+    builder.with_date_of_birth()
+    builder.with_gender()
+    builder.with_postal_address()
+    builder.with_structured_postal_address()
+    builder.with_nationality()
+    builder.with_phone_number()
+    builder.with_selfie()
+    builder.with_email()
+    policy = builder.build()
+
+    attr_names = [attr["name"] for attr in policy["wanted"]]
+    assert len(policy["wanted"]) == 11
+    assert config.ATTRIBUTE_FAMILY_NAME in attr_names
+    assert config.ATTRIBUTE_GIVEN_NAMES in attr_names
+    assert config.ATTRIBUTE_FULL_NAME in attr_names
+    assert config.ATTRIBUTE_DATE_OF_BIRTH in attr_names
+    assert config.ATTRIBUTE_GENDER in attr_names
+    assert config.ATTRIBUTE_POSTAL_ADDRESS in attr_names
+    assert config.ATTRIBUTE_STRUCTURED_POSTAL_ADDRESS in attr_names
+    assert config.ATTRIBUTE_NATIONALITY in attr_names
+    assert config.ATTRIBUTE_PHONE_NUMBER in attr_names
+    assert config.ATTRIBUTE_SELFIE in attr_names
+    assert config.ATTRIBUTE_EMAIL_ADDRESS in attr_names
+
+
+def test_build_with_age_derived_attributes():
+    builder = DynamicPolicyBuilder()
+    builder.with_age_over(18)
+    builder.with_age_under(30)
+    builder.with_age_under(40)
+    policy = builder.build()
+
+    attrs = [attr["derivation"] for attr in policy["wanted"]]
+    assert len(attrs) == 3
+    assert config.ATTRIBUTE_AGE_OVER + "18" in attrs
+    assert config.ATTRIBUTE_AGE_UNDER + "30" in attrs
+    assert config.ATTRIBUTE_AGE_UNDER + "40" in attrs
+
+
+def test_a_derivation_can_exist_only_once():
+    policy = DynamicPolicyBuilder().with_age_under(30).with_age_under(30).build()
+
+    assert len(policy["wanted"]) == 1
+    assert config.ATTRIBUTE_AGE_UNDER + "30" in [
+        a["derivation"] for a in policy["wanted"]
+    ]
+
+
+def test_wanted_auth_types():
+    policy = (
+        DynamicPolicyBuilder()
+        .with_selfie_auth()
+        .with_pin_auth()
+        .with_wanted_auth_type(99)
+        .build()
+    )
+
+    assert len(policy["wanted_auth_types"]) == 3
+    assert DynamicPolicyBuilder.SELFIE_AUTH_TYPE in policy["wanted_auth_types"]
+    assert DynamicPolicyBuilder.PIN_AUTH_TYPE in policy["wanted_auth_types"]
+    assert 99 in policy["wanted_auth_types"]
+
+
+def test_build_with_auth_types_true():
+    policy = DynamicPolicyBuilder().with_selfie_auth(False).build()
+
+    assert len(policy["wanted_auth_types"]) == 0
+
+
+def test_auth_types_can_exist_only_once():
+    policy = (
+        DynamicPolicyBuilder()
+        .with_selfie_auth(True)
+        .with_selfie_auth(False)
+        .with_pin_auth()
+        .with_pin_auth()
+        .build()
+    )
+
+    assert len(policy["wanted_auth_types"]) == 1
```

Comment:
@echarrod we don't have coverage for this situation in the Node SDK (I think I raised this whilst copying Java) - the Node SDK will not allow you to add then remove the same auth when building a policy - is this something to standardise one way or another? I believe [Java is the same](https://github.com/getyoti/yoti-java-sdk/blob/master/yoti-sdk-impl/src/main/java/com/yoti/api/client/shareurl/policy/SimpleDynamicPolicyBuilder.java#L186)

Reply:
The difference in behaviour is here, where the auth type is always added, then filtered out later:
```python
    def with_wanted_auth_type(self, wanted_auth_type, wanted=True):
        self.__wanted_auth_types[wanted_auth_type] = wanted
        return self
```

(I think I prefer this approach, but do we update other SDKs to follow suit?)

Reply:
Good spot, have created SDK-1093 to address this